File: packages/schemas/python/weather.py
from __future__ import annotations
from typing import Literal, Optional
from pydantic import BaseModel

FlightCategory = Literal["VFR", "MVFR", "IFR", "LIFR"]


class MetarObservation(BaseModel):
    icao: str
    raw_text: Optional[str] = None
    observation_time: Optional[str] = None
    temp_c: Optional[float] = None
    dewpoint_c: Optional[float] = None
    wind_dir_deg: Optional[int] = None
    wind_speed_kt: Optional[float] = None
    wind_gust_kt: Optional[float] = None
    visibility_sm: Optional[float] = None
    ceiling_ft: Optional[int] = None
    altimeter_in: Optional[float] = None
    flight_category: Optional[FlightCategory] = None
    wx_string: Optional[str] = None

File: packages/schemas/src/weather.ts
// ── Weather / METAR schemas ───────────────────────────────────────────────────

export type FlightCategory = "VFR" | "MVFR" | "IFR" | "LIFR"

export interface MetarObservation {
  icao: string
  raw_text?: string
  observation_time?: string          // ISO-8601
  temp_c?: number
  dewpoint_c?: number
  wind_dir_deg?: number
  wind_speed_kt?: number
  wind_gust_kt?: number
  visibility_sm?: number
  ceiling_ft?: number
  altimeter_in?: number
  flight_category?: FlightCategory
  wx_string?: string                 // e.g. "-RA BR"
}

export interface MetarResponse {
  data: MetarObservation[]
  cached_at: string                  // ISO-8601
}
